Error when compile with gcc 9

Hi admin,

### OS : 
Linux heradev 5.4.0-66-generic #74-Ubuntu SMP Wed Jan 27 22:54:38 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

### GCC/G++ : 
gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

### error:

/mnt/data/sources/new_install/corokafka/corokafka/impl/corokafka_consumer_manager_impl.cpp: In static member function ‘static Bloomberg::corokafka::MessageBatch Bloomberg::corokafka::ConsumerManagerImpl::messageBatchReceiveTask(Bloomberg::corokafka::ConsumerTopicEntry&, Bloomberg::corokafka::IoTracker)’:
/mnt/data/sources/new_install/corokafka/corokafka/impl/corokafka_consumer_manager_impl.cpp:977:52: error: missing template arguments before ‘(’ token
  977 |                                  std::move_iterator(tempBatch.begin()),
      |                                                    ^
/mnt/data/sources/new_install/corokafka/corokafka/impl/corokafka_consumer_manager_impl.cpp:978:52: error: missing template arguments before ‘(’ token
  978 |                                  std::move_iterator(tempBatch.end()));
      |                                                    ^
make[2]: *** [corokafka/CMakeFiles/corokafka.dir/build.make:310: corokafka/CMakeFiles/corokafka.dir/impl/corokafka_consumer_manager_impl.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:94: corokafka/CMakeFiles/corokafka.dir/all] Error 2
make: *** [Makefile:130: all] Error 2

--- a/corokafka/impl/corokafka_consumer_manager_impl.cpp
+++ b/corokafka/impl/corokafka_consumer_manager_impl.cpp
@@ -966,6 +966,7 @@ MessageBatch ConsumerManagerImpl::messageBatchReceiveTask(ConsumerTopicEntry& en
         else {
             //break the call into smaller chunks so we don't block permanently if shutting down
             while (!entry._interrupt && ((ssize_t)batch.size() < entry._readSize)) {
+                using IterType = MessageBatch::iterator;
                 MessageBatch tempBatch = entry._consumer->poll_batch(entry._readSize, entry._minPollInterval);
                 if (batch.empty()) {
                     std::swap(batch, tempBatch);
@@ -974,8 +975,8 @@ MessageBatch ConsumerManagerImpl::messageBatchReceiveTask(ConsumerTopicEntry& en
                     //merge batches
                     batch.reserve(entry._readSize);
                     batch.insert(batch.end(),
-                                 std::move_iterator(tempBatch.begin()),
-                                 std::move_iterator(tempBatch.end()));
+                                 std::move_iterator<IterType>(tempBatch.begin()),
+                                 std::move_iterator<IterType>(tempBatch.end()));
                 }
             }
         }

--- a/corokafka/impl/corokafka_received_message_impl.h
+++ b/corokafka/impl/corokafka_received_message_impl.h
@@ -154,7 +154,11 @@ ReceivedMessageImpl<KEY,PAYLOAD,HEADERS>::~ReceivedMessageImpl()
         // auto-persistence is turned off or the offset has been persisted manually
         return;
     }
+#if (__cplusplus < 201703L)
     if (std::uncaught_exception() && !_offsetSettings._autoOffsetPersistOnException) {
+#else
+    if ((std::uncaught_exceptions() != 0) && !_offsetSettings._autoOffsetPersistOnException) {
+#endif
         return; // don't commit if we are being destroyed as the result of an exception
     }
     doCommit(_offsetSettings._autoCommitExec);

